File: BookSuggestionSystem/java/BookSuggestionSystem.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Random;

public class BookSuggestionSystem {

    public static void displayMenu() {
    
       System.out.println("""
            1. Get Suggestions
            2. Add Book
            3. Remove Book
            4. Update Book
            5. Show All Books
            """);

    }


    public static String getBookSuggestion (ArrayList <String> books, Random bookSuggestion) {
    
        int randomIndex = 0;
        String selectedBook = "";

        if(books.size() == 0) {
            throw new IllegalArgumentException("Your library is empty. please add books");
        }

        else {
            for(int index = 0; index < books.size(); index++) {
                randomIndex = bookSuggestion.nextInt(books.size());
            }

            selectedBook = books.get(randomIndex);
        }

        return selectedBook;
    }


    public static boolean isPageSuggestionWithinRange (Random pageSuggestion) {

        int randomPageNumber = pageSuggestion.nextInt(1, 101);

        if(randomPageNumber > 0 && randomPageNumber <= 100) return true;

        return false;

    }


    public static String addBook (String newBook, ArrayList <String> books) {
        
        for(String book : books) {
            if(book.equalsIgnoreCase(newBook)) {
                throw new IllegalArgumentException("Book already exists");
            }
        }             

        books.add(newBook);
        return "Book added successfully"; 
     
    }


    public static String removeBook (String oldBook, ArrayList <String> books) {

        for(String book : books) {
            if(book.equalsIgnoreCase(oldBook)) {
                books.remove(oldBook);
                return "Book removed successfully";      
            }
        }
  
        throw new IllegalArgumentException("Book not found");

    }


    public static String updateBook (String oldTitle, String newTitle, ArrayList <String> books) {

        for(String book : books) {
            if(!book.equalsIgnoreCase(oldTitle)) {
                throw new IllegalArgumentException("Book not found");
            }
        }             
        for(String book : books) {
            if(book.equalsIgnoreCase(oldTitle)) {
                books.remove(book);
                books.add(newTitle);
            }
        } 
 
        return "Book updated successfully"; 
     
    }


    public static void showListOfBooks(ArrayList <String> books) {

        for (String book : books) {
            System.out.println(book);
        }

    }
    

}

File: BookSuggestionSystem/python/booksuggestionsystem.py
import random

def display_menu():
    
    print("""
        1. Get Suggestions
        2. Add Book
        3. Remove Book
        4. Update Book
        5. Show All Books
        """)


def get_book_suggestion (books):

    if len(books) == 0: 
        raise ValueError("Your library is empty. please add books")   

    else: 
        for index in range (len(books)): 
            selected_book = random.choice(books)
        
    return selected_book


def is_page_suggestion_within_range():

    random_page_number = random.randint(1, 101)

    if random_page_number > 0 and random_page_number <= 100: 
        return True

    return False


def add_book (new_book, books): 
        
    for book in books: 
        if book.lower() == new_book.lower(): 
            raise ValueError("Book already exists")                

    books.append(new_book)
    return "Book added successfully" 
     

def remove_book (old_book, books): 

    for book in books:
        if book.lower() == old_book.lower(): 
            books.remove(book)
            return "Book removed successfully"        
  
    raise ValueError("Book not found")


def update_book (old_title, new_title, books): 

    for book in books: 
        if book.lower() != old_title.lower(): 
            raise ValueError("Book not found")       
                 
    for book in books: 
        if book.lower() == old_title.lower(): 
            books.remove(book)
            books.append(new_title)    
 
    return "Book updated successfully" 
     

def show_list_of_books(books): 

    for book in books: 
        print(book)
    

